src/ray/gcs/gcs_client/global_state_accessor.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <boost/algorithm/string.hpp>
+
+#include "global_state_accessor.h"
+
+namespace ray {
+namespace gcs {
+
+GlobalStateAccessor::GlobalStateAccessor(const std::string &redis_address,
+                                         const std::string &redis_password,
+                                         bool is_test) {
+  RAY_LOG(INFO) << "Redis server address = " << redis_address
+                << ", is test flag = " << is_test;
+  std::vector<std::string> address;
+  boost::split(address, redis_address, boost::is_any_of(":"));
+  RAY_CHECK(address.size() == 2);
+  GcsClientOptions options;
+  options.server_ip_ = address[0];
+  options.server_port_ = std::stoi(address[1]);
+  options.password_ = redis_password;
+  options.is_test_client_ = is_test;
+  gcs_client_.reset(new ServiceBasedGcsClient(options));
+
+  io_service_.reset(new boost::asio::io_service());
+
+  thread_io_service_.reset(new std::thread([this] {
+    std::unique_ptr<boost::asio::io_service::work> work(
+        new boost::asio::io_service::work(*io_service_));
+    io_service_->run();
+  }));
+}
+
+bool GlobalStateAccessor::Connect() { return gcs_client_->Connect(*io_service_).ok(); }
+
+void GlobalStateAccessor::Disconnect() {
+  gcs_client_->Disconnect();
+  io_service_->stop();
+  thread_io_service_->join();
+}
+
+std::vector<std::string> GlobalStateAccessor::GetJobTable() {
```

Comment:
We can just return `vector<JobTableData>` here. And in Cython, we can parse C++ `JobTableData` to the final Python dict.

Reply:
Now gcs_utils.JobTableData.FromString can deserialize string of JobTableData directly, which is easier.